core/src/main/java/com/sopt/core/designsystem/component/timetable/NoostakTimeTable.kt

```diff
@@ -0,0 +1,188 @@
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.unit.dp
+import com.sopt.core.designsystem.theme.NoostakTheme
+import com.sopt.domain.entity.TimeTableEntity
+import java.time.LocalDate
+import java.time.format.DateTimeFormatter
+import java.time.format.TextStyle
+import java.util.Locale
+
+@Composable
+fun NoostakTimeTable(
+    data: TimeTableEntity,
+    modifier: Modifier = Modifier
+) {
+    val days = data.timeEntity.size
+    val timeSlots = calculateTimeSlots(data.startTime, data.endTime)
+
+    LazyVerticalGrid(
+        modifier = modifier
+            .border(
+                width = 1.dp,
+                color = NoostakTheme.colors.gray200,
+                shape = RoundedCornerShape(8.dp)
+            ),
+        columns = GridCells.Fixed(days + 1)
+    ) {
+        items((days + 1) * (timeSlots + 1)) { index ->
+            val rowIndex = index / (days + 1)
+            val columnIndex = index % (days + 1)
+
+            val cellType = determineCellType(rowIndex, columnIndex)
+            val backgroundColor = getBackgroundColor(cellType, rowIndex, columnIndex, data)
+            val text = getCellText(cellType, rowIndex, columnIndex, data)
+
+            NoostakTimeTableBox(
+                index = index,
+                days = days,
+                timeSlots = timeSlots,
+                backgroundColor = backgroundColor,
+                text = text
+            )
+        }
+    }
+}
+
+@Composable
+fun getBackgroundColor(
+    cellType: CellType,
+    rowIndex: Int,
+    columnIndex: Int,
+    data: TimeTableEntity
+): Color {
+    return when (cellType) {
+        CellType.Blank, CellType.DateHeader, CellType.TimeHeader -> Color.Transparent
+        CellType.Data -> {
+            val startHour = data.startTime.split(":")[0].toInt()
+            val currentHour = startHour + (rowIndex - 1)
+            val availableTimes = data.timeEntity.getOrNull(columnIndex - 1)?.times
+
+            val matchingLevel = availableTimes?.find { timeEntity ->
+                val entityStartHour = timeEntity.startTime.split(":")[0].toInt()
+                val entityEndHour = timeEntity.endTime.split(":")[0].toInt()
+                currentHour in entityStartHour until entityEndHour
+            }?.level
+
+            getColorByLevel(matchingLevel ?: 0)
+        }
+    }
+}
+
+fun getCellText(
+    cellType: CellType,
+    rowIndex: Int,
+    columnIndex: Int,
+    data: TimeTableEntity
+): String {
+    val startHour = data.startTime.split(":")[0].toInt()
+
+    return when (cellType) {
+        CellType.Blank -> "\n"
+        CellType.DateHeader -> {
+            val dateEntity = data.timeEntity.getOrNull(columnIndex - 1)
+            val date = dateEntity?.date ?: ""
+            formatDateHeader(date)
+        }
+
+        CellType.TimeHeader -> "${startHour + (rowIndex - 1)}시"
+        CellType.Data -> ""
+    }
+}
```

Comment:
P2: 이렇게 변수 없이 깔끔하게 쓰는 것도 좋을 것 같네요! (가독성 부분에서는 기존 코드가 더 좋은 것 같기도 합니다ㅏㅏ)

```suggestion
fun getCellText(
    cellType: CellType,
    rowIndex: Int,
    columnIndex: Int,
    data: TimeTableEntity
): String = when (cellType) {
    CellType.Blank -> "\n"
    CellType.DateHeader -> formatDateHeader(data.timeEntity.getOrNull(columnIndex - 1)?.date ?: "")
    CellType.TimeHeader -> "${data.startTime.split(":")[0].toInt() + (rowIndex - 1)}시"
    CellType.Data -> ""
}

```